src/main/java/org/testng/internal/MethodInvocationHelper.java

```diff
@@ -288,7 +288,6 @@ private static void invokeWithTimeoutWithNewExecutor(ITestNGMethod tm, Object in
       ThreadTimeoutException exception = new ThreadTimeoutException("Method "
           + tm.getQualifiedName() + "()"
           + " didn't finish within the time-out " + realTimeOut);
-      exception.setStackTrace(exec.getStackTraces()[0]);

```

Comment:
> But it doesn't explain why ThreadFactoryImpl doesn't feed its threads list.

Because new threads are created in `org.testng.internal.thread.TestNGThreadFactory` and the base class doesn't have visibility to the `org.testng.internal.thread.ThreadUtil.ThreadFactoryImpl#threads` data member. I have now fixed this.